crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -232,6 +236,10 @@ impl Diagnostic {
     pub fn primary_tags(&self) -> Option<&[DiagnosticTag]> {
         self.primary_annotation().map(|ann| ann.tags.as_slice())
     }
+
+    pub fn expect_span(&self) -> Span {
```

Comment:
Can you add a comment documenting this? In particular, it would be good to state the panic conditions and the higher level motivation for why you might want to use this. (e.g., "When working with diagnostics in Ruff, the context requires that every diagnostic have a primary span.")

I think I'd also rename this to `expect_primary_span()` so that it's consistent with `primary_span()`.